prov/verbs/src/fi_verbs.c

```diff
@@ -815,25 +792,63 @@ static int fi_ibv_init_info(const struct fi_info *hints)
 	default:
 		FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unknown transport type");
 		ret = -FI_ENODATA;
-		goto err3;
+		goto err;
 	}
 
 	if (!(fi->domain_attr->name = strdup(ctx->device->name))) {
 		ret = -FI_ENOMEM;
-		goto err3;
+		goto err;
 	}
 
-	verbs_info = fi;
-	rdma_free_devices(ctx_list);
-unlock:
-	pthread_mutex_unlock(&verbs_info_lock);
+	*info = fi;
 	return 0;
-
-err3:
+err:
 	fi_freeinfo(fi);
-err2:
+	return ret;
+}
+
+static int fi_ibv_init_info(const struct fi_info *hints)
```

Comment:
Something is not right here.  This call is initializing a list of fi_info structures.  The initialization should be completely independent of any 'hints'.